diff --git a/lib/std/Io/Threaded.zig b/lib/std/Io/Threaded.zig
index 8ca455c..49e2747 100644
--- a/lib/std/Io/Threaded.zig
+++ b/lib/std/Io/Threaded.zig
@@ -15214,7 +15214,7 @@ fn processReplace(userdata: ?*anyopaque, options: process.ReplaceOptions) proces
         });
     };
 
-    if (is_darwin) if (t.spawnDarwin(
+    if (is_darwin and process.can_spawn) if (t.spawnDarwin(
         .replace,
         options.exe,
         argv_buf.ptr,
